cmd/util/ledger/util/snapshot/payload_snapshot.go

```diff
@@ -0,0 +1,266 @@
+package snapshot
+
+import (
+	"sort"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/fvm/storage/snapshot"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type MigrationSnapshotType int
+
+const (
+	// MapBased is more efficient when the number of payloads that are going to change is either 0
+	// or large compared to the total number of payloads (more than 30% of the current number of payloads).
+	MapBased MigrationSnapshotType = iota
+	// IndexMapBased is more efficient when the number of payloads that are going to change is small
+	// compared to the total number of payloads (less than 30% of the current number of payloads).
+	IndexMapBased
```

Comment:
nit: `MapBased` and `IndexMapBased` can be confusing to differentiate without reading comments.  Maybe we can rename them.

Reply:
I didn't have any good ideas I renamed them according to suggested usage. Let me know what you think.

Reply:
Maybe something like this to help user to choose the snapshot type (without implementation detail):
- `ReadOnlySnapshotType`
- `LargeChangeSetSnapshotType`
- `SmallChangeSetSnapshotType`

Both `ReadOnlySnapshotType` and `LargeChangeSetSnapshotType` create `MapBased` snapshot. 

`SmallChangeSetSnapshotType` creates `IndexMapBased` snapshot.

This just an idea, so please feel free to choose any name you prefer.